src/java.base/share/classes/sun/net/ftp/impl/FtpClient.java

```diff
@@ -111,16 +110,13 @@
 
     static {
         final int vals[] = {0, 0};
-        final String encs[] = {null};
-
-        AccessController.doPrivileged(
-                new PrivilegedAction<Object>() {
-
-                    public Object run() {
+        @SuppressWarnings("removal")
+        final String enc = AccessController.doPrivileged(
+                new PrivilegedAction<String>() {
+                    public String run() {
                         vals[0] = Integer.getInteger("sun.net.client.defaultReadTimeout", 300_000).intValue();
                         vals[1] = Integer.getInteger("sun.net.client.defaultConnectTimeout", 300_000).intValue();
-                        encs[0] = System.getProperty("file.encoding", "ISO8859_1");
-                        return null;
+                        return System.getProperty("file.encoding", "ISO8859_1");
                     }
```

Comment:
It is a bit strange that "file.encoding" seem to get a special treatment - but I guess that's OK.

Reply:
You might say we thus avoid wasting the return value, as much as possible.

Reply:
Hi Max - yes - I believe I'd prefer that than potentially missing more removal warnings  because the scope is too large. And with the refactoring you did with using lambdas it actually doesn't look so bad :-)